compiler/cpp/src/generate/t_swift_3_generator.cc

```diff
@@ -0,0 +1,2458 @@
+ *
+ * @param tstruct The struct definition
+ */
+void t_swift_3_generator::generate_xception(t_struct* txception) {
+  generate_swift_struct(f_decl_, txception, false, false);
+  generate_swift_struct_implementation(f_impl_, txception, false, false);
+}
+
+void t_swift_3_generator::generate_docstring(ofstream& out,
+                                             string& doc) {
+  if (doc != "") {
+    std::vector<std::string> strings;
+
+    std::string::size_type pos = 0;
+    std::string::size_type prev = 0;
+    while ((pos = doc.find("\n", prev)) != std::string::npos)
+    {
+        strings.push_back(doc.substr(prev, pos - prev));
+        prev = pos + 1;
+    }
+
+    // To get the last substring (or only, if delimiter is not found)
+    strings.push_back(doc.substr(prev));
+
+    vector<string>::const_iterator d_iter;
+    for (d_iter = strings.begin(); d_iter != strings.end(); ++d_iter) {
+      out << indent() << "/// " << (*d_iter) << endl;
+    }
+  }
+}
+
+
+
+/**
+ * Generate the interface for a struct. Only properties and
+ * init methods are included.
+ *
+ * @param tstruct The struct definition
+ * @param is_private
+ *                Is the struct public or private
+ */
+void t_swift_3_generator::generate_swift_struct(ofstream& out,
+                                              t_struct* tstruct,
+                                              bool is_private,
+                                              bool is_result) {
+
+
+  string doc = tstruct->get_doc();
+  if (doc != "") {
+    std::vector<std::string> strings;
+
+    std::string::size_type pos = 0;
+    std::string::size_type prev = 0;
+    while ((pos = doc.find("\n", prev)) != std::string::npos)
+    {
+        strings.push_back(doc.substr(prev, pos - prev));
+        prev = pos + 1;
+    }
+
+    // To get the last substring (or only, if delimiter is not found)
+    strings.push_back(doc.substr(prev));
+
+    vector<string>::const_iterator d_iter;
+    for (d_iter = strings.begin(); d_iter != strings.end(); ++d_iter) {
+      out << "/// " << (*d_iter) << endl;
+    }
+  }
+
+  // properties
+  const vector<t_field*>& members = tstruct->get_members();
+  vector<t_field*>::const_iterator m_iter;
+
+
+  if (tstruct->is_union()) {
+    // special, unions
+    out << indent() << "public enum " << tstruct->get_name();
+    block_open(out);
+    for (m_iter = members.begin(); m_iter != members.end(); ++m_iter) {
+      out << endl;
+      // TODO: Defaults
+      // add swift docstring if available
+      string doc = (*m_iter)->get_doc();
+      if (doc != "") {
+        out << indent() << "/// " << doc;
+      }
+      out << indent() << "case "
+          << maybe_escape_identifier((*m_iter)->get_name()) << "(val: "
+          << type_name((*m_iter)->get_type(), false) << ")" << endl;
+    }
+  } else {
+    // Normal structs
+
+    string visibility = is_private ? "fileprivate" : "public";
+
+    out << indent() << visibility << " final class " << tstruct->get_name();
```

Comment:
I think it would be a more natural choice to use Swift structs for Thrift structs, since they are the same thing conceptually. Did you have any particular reason for choosing reference types instead? There is a performance argument, as reference counting will be more performant if the struct contains lots of other reference types.